drivers/soundwire/intel_init.c

```diff
@@ -131,6 +131,7 @@ sdw_intel_scan_controller(struct sdw_intel_acpi_info *info)
 	dev_dbg(&adev->dev, "ACPI reports %d SDW Link devices\n", count);
 
 	info->count = count;
+	info->link_mask = 0;
```

Comment:
out of curiosity I did trace back `info` to `hda_dsp_probe()` where it's allocated, using `kzalloc()`, so, initially `.link_mask = 0`. Is this change just a precaution / clean up to make sure this function doesn't depend on external conditions or can it indeed be called multiple times?

Reply:
@lyakh Probably not in practise, but sdw_intel_acpi_scan() is a public exported function, so would seem safer to not assume a zero'ed out info struct.

Reply:
@lyakh in this case the function is called from the legacy probe as well, and the info variable is allocated on the stack, so not zeroed out. I could have used a memset() but I think it's just safer in general to zero a variable before loop-ORing it.